homeassistant/components/tradfri/base_class.py

```diff
@@ -34,22 +42,28 @@ class TradfriBaseClass(Entity):
 
     _attr_should_poll = False
 
-    def __init__(self, device, api, gateway_id):
+    def __init__(
+        self, device: Command, api: Callable[[str], Any], gateway_id: str
+    ) -> None:
         """Initialize a device."""
         self._api = handle_error(api)
-        self._device = None
-        self._device_control = None
-        self._device_data = None
+        self._device: Command | None = None
+        self._device_control: SocketControl | None = None
+        self._device_data: Socket | Light | Blind | None = None
         self._gateway_id = gateway_id
         self._refresh(device)
 
     @callback
-    def _async_start_observe(self, exc=None):
+    def _async_start_observe(self, exc: Exception | None = None) -> None:
         """Start observation of device."""
+        if not self._device:
```

Comment:
It should be enough to assert `self._device` since `_refresh` is called in init and we never set `self._device` to `None` after that.

Maybe we can move the initialization to `_refresh` and type `self._device` there instead? I'm not sure if any of our linters will complain.

Reply:
I solved it a little bit differently, but thanks for the idea, it saves a lot of code.